scapy/fields.py

```diff
@@ -309,6 +309,27 @@ def _evalcond(self, pkt):
         # type: (Packet) -> bool
         return bool(self.cond(pkt))
 
+    def any2i(self, pkt, x):
+        # type: (Optional[Packet], Any) -> Any
+        # BACKWARD COMPATIBILITY
+        # Note: we shouldn't need this function. (it's not correct)
+        # However, having i2h implemented (#2364), it changes the default
+        # behavior and broke all packets that wrongly use two ConditionalField
+        # with the same name. Those packets are the problem: they are wrongly
+        # built (they should either be re-using the same conditional field, or
+        # using a MultipleTypeField).
+        # But I don't want to dive into fixing all of them just yet,
+        # so for now, let's keep this this way, even though it's not correct.
+        if type(self.fld) is Field:
```

Comment:
Codacity complains about this line too =)

Reply:
Except they're wrong. I only want Field to be detected, not any of its subfields.